src/pytorch_lightning/core/saving.py

```diff
@@ -58,11 +61,13 @@ class ModelIO:
     def load_from_checkpoint(
         cls,
         checkpoint_path: Union[str, IO],
-        map_location: Optional[Union[Dict[str, str], str, torch.device, int, Callable]] = None,
+        map_location: Optional[
+            Union[str, Callable, torch.device, Dict[Union[str, torch.device], Union[str, torch.device]]]
+        ] = None,
         hparams_file: Optional[str] = None,
         strict: bool = True,
-        **kwargs,
-    ):
+        **kwargs: Any,
+    ) -> Union["ModelIO", "pl.LightningModule", "pl.LightningDataModule"]:
```

Comment:
shouldn't the return type here be the same as from `_load_from_checkpoint`, i.e., just `Union["pl.LightningModule", "pl.LightningDataModule"]`

Reply:
updated in https://github.com/Lightning-AI/lightning/pull/13932/commits/99643a0ed8ad27dd5bc94a5eaf7070523223b5dd